### ISSUE TITLE
lint `let ref p: str = *"hello";`

### ISSUE THREAD

**Issue by oli-obk** [Manishearth/rust-clippy#1431]
Is there ever a reason to have an unsized type and ref it instead of having a reference to an unsized type?

I think we should simply suggest to use `let p: &str = &*"hello";` instead (and make sure our quoting code erases `&*` or sth)

**Comment by mcarton** [comment]
We already lint `let ref`:

```rust
fn main() {
    let ref _foo = *"hello";
}
```

```rust
warning: `ref` on an entire `let` pattern is discouraged, take a reference with `&` instead, #[warn(toplevel_ref_arg)] on by default
 --> src/main.rs:2:9
  |
2 |     let ref _foo = *"hello";
  |         ^^^^^^^^
  |
help: try
  |     let _foo = &(*"hello");
  = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#toplevel_ref_arg
```

**Comment by oli-obk** [comment]
oh ^^ well, only improving the suggestions left then...
